src/DynamoPackages/Package.cs

```diff
@@ -572,21 +575,66 @@ internal void SetAsLoaded()
             RaisePropertyChanged(nameof(LoadState));
         }
 
+        AssemblyLoadContext GetPackageALC()
+        {
+            if (LoadedAssemblies.Any())
+            {
+                if (LoadedAssemblies.All(x => AssemblyLoadContext.GetLoadContext(x.Assembly) != AssemblyLoadContext.Default))
+                {
+                    return AssemblyLoadContext.GetLoadContext(LoadedAssemblies[0].Assembly);
+                }
+            }
+            return AssemblyLoadContext.Default;
+        }
+
+
+
         internal void UninstallCore(CustomNodeManager customNodeManager, PackageLoader packageLoader, IPreferences prefs)
         {
             if (LoadedAssemblies.Any())
             {
+                bool unloadAnyPackage = DynamoModel.FeatureFlags?.CheckFeatureFlag("UnloadAnyIsolatedPackage", false) ?? false;
+                bool unloadThisPackage = DynamoModel.FeatureFlags?.CheckFeatureFlag("UnloadIsolatedPackage_" + this.Name, false) ?? false;
+
+                if (unloadAnyPackage || unloadThisPackage)
+                {
+                    var alc = GetPackageALC();
+                    if (alc != AssemblyLoadContext.Default)
+                    {
+                        void Alc_Unloading(AssemblyLoadContext obj)
+                        {
+                            alc.Unloading -= Alc_Unloading;
+                            try
+                            {
+                                RemovePackage(customNodeManager, packageLoader, prefs);
+                            }
+                            catch
+                            {
+                                // Exception already logged inside RemovePackage
+                            }
+                        }
+
+                        alc.Unloading += Alc_Unloading;
+                        alc.Unload();
+                    }
+                }
+
                 MarkForUninstall(prefs);
                 return;
             }
 
+            RemovePackage(customNodeManager, packageLoader, prefs);
+        }
+
+        private void RemovePackage(CustomNodeManager customNodeManager, PackageLoader packageLoader, IPreferences prefs)
```

Comment:
I'd think you'd also need to force a GC in DS and .net?

Reply:
I'll try that